lib/utils/logger.js

```diff
@@ -11,38 +11,34 @@
 const debug = require('debug');
 
 class Logger {
+
+  static getLogger() {
+    if (this.logger) {
+      return this.logger;
+    }
+    return this.logger = new Logger();
+  }
+
+  static warn(message) {
+    return this.getLogger().warn(message);
+  }
+
   constructor(config) {
 
     this.config = Object.assign({
       context: 'sequelize',
       debug: true
     }, config);
-
-    this.debug = debug(this.config.context);
-  }
-
-  debug(message) {
-    if (this.config.debug) {
-      this.debug(message);
-    }
   }
 
   warn(message) {
     // eslint-disable-next-line no-console
     console.warn(`(${this.config.context}) Warning: ${message}`);
   }
 
-  debugContext(childContext) {
-    if (!childContext) {
-      throw new Error('No context supplied to debug');
```

Comment:
Can probably let it slide, but this empty check may prevent logging to `sequelize:undefined` etc